src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,55 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier randomColor = new ColorSupplier();

```

Comment:
Don't use verbs for class/interface names. The name 'ColorSupplier' suggests it could be a method rather than a class. Consider renaming it to something like 'ColorProvider' or 'ColorGenerator' to reflect that it's an object.